api/src/main/java/org/openmrs/Obs.java

```diff
@@ -90,13 +112,17 @@
 	private static final String FORM_NAMESPACE_PATH_SEPARATOR = "^";
 	
 	private static final int FORM_NAMESPACE_PATH_MAX_LENGTH = 255;
-	
+	@Id
+	@GeneratedValue(strategy = GenerationType.SEQUENCE)
```

Comment:
I think AUTO generator type is a better option, it lets hibernate decide the best based on the underlying DB

Reply:
I had used the default used in the hbm.xml file originally...so should i change to auto then?